packages/gatsby/src/redux/run-sift.js

```diff
@@ -316,9 +311,9 @@ const filterWithoutSift = (filter, nodeTypeNames, typedKeyValueIndexes) => {
  *   will be limited to 1 if `firstOnly` is true
  */
 const filterWithSift = (filter, firstOnly, nodeTypeNames, resolvedFields) => {
-  let nodes = []
-
-  nodeTypeNames.forEach(typeName => addResolvedNodes(typeName, nodes))
+  const nodes = _.flatten(
+    nodeTypeNames.map(typeName => addResolvedNodes(typeName))
+  )
```

Comment:
I'll be honest, not thrilled with the `_.flatten`. We're changing a simple JS approach to replace it with an abstraction with overhead and I'm not sure the reason is worth it. I'd rather have less third party functions than more.

That said, we should bench it. If it turns out faster (things at scale change the rule, nothing is obvious) then we should do it.

```suggestion
const nodes = [].concat(
  ...nodeTypeNames.map(typeName => addResolvedNodes(typeName))
)
```

But I'm not sure this is any better for legibility. Or any improvement to perf, for that matter.

Reply:
implemented, it looks cleaner to me